addons/event_sale/views/sale_order_views.xml

```diff
@@ -22,12 +22,12 @@
                     attrs="{'invisible': [('event_ok', '=', False)], 'required': [('event_ok', '!=', False)]}"
                     options="{'no_open': True, 'no_create': True}"
                 />
+                <field name="available_event_ticket_ids" invisible="1"/>
                 <field
                     name="event_ticket_id"
                     domain="[
-                        ('event_id', '=', event_id),
-                        ('product_id','=',product_id),
-                        '|', ('seats_limited', '=', False), ('seats_available', '>', 0), '|', ('company_id', '=', False), ('company_id', '=', parent.company_id)
+                        ('event_id', '=', event_id), ('product_id','=',product_id), ('id', 'in', available_event_ticket_ids),
```

Comment:
warning for a big event it can create a big domain.

cc @rco-odoo

Reply:
Indeed, not a good idea.